### ISSUE TITLE
Make aria-label mandatory for buttons

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/nextcloud-vue#4214]
We've logged a warning for a while

> You need to fill either the text or the ariaLabel props in the button component

I think it would be fair to start enforcing this prop in a future major version.

Ref https://github.com/nextcloud/nextcloud-vue/commit/d441b9dfb47aebae17bae4c21493057e4a7c862c @marcoambrosini

**Comment by susnux** [comment]
From the discussion [here](https://github.com/nextcloud/photos/pull/1906) it seems we should set the aria label always to the visual text and use a different attribute (`title`? `aria-description`?) for more context.

**Comment by dvaerum** [comment]
This is still something one gets warnings about in the console of the browser in Nextcloud 29.0.6, does anyone know what the status is for this?

**Comment by ShGKme** [comment]
Making this prop required doesn't fix the issue. It would only change `You need to fill either the text or the ariaLabel` warning to the default Vue warning about missing the required prop.

At the same time, `aria-label` cannot be required. It should be set only when there is no text content of the button. Having this prop required may also result in different value in the text and `aria-label`.

The issue should be solved on the app developers side. They must provide aria-label when a button has no text.

One proposal I may suggest is to make it easier - use `title` attribute as a fallback value for `aria-label`. This fixes some cases, but might not always be correct.

**Comment by susnux** [comment]
> One proposal I may suggest is to make it easier - use title attribute as a fallback value for aria-label. This fixes some cases, but might not always be correct.

I understand the idea but agree more with the last part "not always correct". The title is often only a description and not a correct label. So I think this would hide the real problem. Instead this is a bug in the app, the app developer should add `aria-label` for all icon-only buttons.

**Comment by marcoambrosini** [comment]
I think that the status quo with the warning is reasonable. Maybe we could make the warning into an error so there are more chances it gets picked up?

**Comment by dvaerum** [comment]
> The issue should be solved on the app developers side. They must provide aria-label when a button has no text.

This makes sense, but is there a way to make it more obvious which app is forgetting to add the `aria-label`? because currently it is not obvious to me there I should go and submit my pull request. Or maybe there already is a great way to figure this out and I just don't know about it :sweat_smile:

**Comment by susnux** [comment]
We could include a reference to the HTML node in the log entry so you can click on it in dev tools to spot it in the browser inspector.
But I do not think we can do more here, enforcing `aria-label` seems to be overkill and might lead to inconsistency between visual and accessible name.
